code/modules/mob/_modifiers/fire.dm

```diff
@@ -0,0 +1,36 @@
+// Ignition, but confined to the modifier system.
+// This makes it more predictable and thus, easier to balance.
+/datum/modifier/fire
+	name = "on fire"
+	desc = "You are on fire! You will be harmed until the fire goes out or you extinguish it with water."
+	mob_overlay_state = "on_fire"
+
+	on_created_text = "<span class='danger'>You combust into flames!</span>"
```

Comment:
![dictionary](https://puu.sh/Fhk7R/b32a192cce.png)
`You [are consumed by fire] into flames!`
Burst, perhaps?